src/languageFeatures/codeActions/extractLinkDef.ts

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken } from 'vscode-languageserver';
+import * as lsp from 'vscode-languageserver-types';
+import * as nls from 'vscode-nls';
+import { URI } from 'vscode-uri';
+import { translatePosition } from '../../types/position';
+import { makeRange, rangeIntersects } from '../../types/range';
+import { getLine, ITextDocument } from '../../types/textDocument';
+import { WorkspaceEditBuilder } from '../../util/editBuilder';
+import { ExternalHref, HrefKind, InternalHref, LinkDefinitionSet, MdDocumentLinksInfo, MdInlineLink, MdLink, MdLinkDefinition, MdLinkKind, MdLinkProvider } from '../documentLinks';
+import { getExistingDefinitionBlock } from '../organizeLinkDefs';
+
+const localize = nls.loadMessageBundle();
+
+export class MdExtractLinkDefinitionCodeActionProvider {
+
+	public static readonly genericTitle = localize('genericTitle', 'Extract to link definition');
+
+	private static kind = lsp.CodeActionKind.RefactorExtract + '.linkDefinition';
```

Comment:
Convert is probably a better fix for this instead of `extract`

Reply:
Actually will stick with extract since it's a well known kind  and the behavior is similar enough to make sense